filesystem.cpp

```diff
@@ -1,64 +1,222 @@
+    for(auto i:currentdir->getSubdirectories()){
+        if(i->getname()==filename){
+            cout<<"Ez a nev mar foglalt"<<endl;
+            return 0;
+        }
+    }
+    currentdir->touch(filename);
+    return 1;
+}
 void filesystem::ls(){
     directory* tmp;
     tmp=currentdir;
     list<string> path;
-    while(tmp->getParent()!=nullptr){
-        path.push_front(tmp->getParent()->getName());
-        tmp=tmp->getParent();
+    path.push_front(currentdir->getname());
+    while(tmp->getparent()!=nullptr){
+        path.push_front(tmp->getparent()->getname());
+        tmp=tmp->getparent();
     }
+    for(auto i:path){
+        cout<<i;
+        cout<<"/";
+    }
+    cout<<endl;
     for(auto i:currentdir->getSubdirectories()){
-        for(auto i:path){
-            cout<<i;
-            cout<<"/";
-        }
-        currentdir->print();
-        i->ls();
+        cout<<"   "<<i->getname();
         cout<<endl;
     }
 }
-//cd mukodik
+
 int filesystem::cd(string to){
     if(to==".."){
-        currentdir=currentdir->getParent();
+        currentdir=currentdir->getparent();
         return 1;
     }
+    int tmp=0;
     for(auto& i:currentdir->getSubdirectories()){
         //i->print();
-        if(i->getName()==to){
-        currentdir=i;
-        return 1;
+        if(i->getname()==to){
+        tmp+=1;
+        directory* temp=dynamic_cast<directory*>(i);
+        if(temp!=nullptr){
+            currentdir=temp;
+            return 1;
+        }
+        else{
+            cout<<"File-ba nem lehet cd-zni"<<endl;
+            return 0;
         }
+        }
+    }
+    if(tmp==0){
+        cout<<"Az adott mappa nem letezik"<<endl;
     }
     return 0;
 }
 
+vector<string> filesystem::split(const string& str, const char& delim)
+{
+    vector<string> tokens;
+    size_t prev = 0, pos = 0;
+    do
+    {
+        pos = str.find(delim, prev);
+        if (pos == string::npos) pos = str.length();
+        string token = str.substr(prev, pos-prev);
+        if (!token.empty()) tokens.push_back(token);
+        prev = pos + 1;
+    }
+    while (pos < str.length() && prev < str.length());
+    return tokens;
+}
+
+int filesystem::absolutepathcd(string path){
+    currentdir=root;
+    vector<string>route=this->split(path,'/');
+    unsigned int flag=1;
+    if(route[0]!="root"){
+        cout<<"Nincs ilyen eleresi utvonal"<<endl;
+        return 0;
+    }
+    for(unsigned int i=1;i<route.size();i++){
+        for(auto j:currentdir->getSubdirectories()){
+            directory* temp=dynamic_cast<directory*>(j);
+            if(temp!=nullptr){
+                if(temp->getname()==route[i]){
+                    currentdir=temp;
+                    flag++;
+                }
+            }
+        }
+    }
+    if(flag==route.size()){
+        return 1;
+    }
+    else{
+        cout<<"Nincs ilyen eleresi utvonal"<<endl;
+        return 0;
+    }
+}
+
+int filesystem::absolutepathmkdir(string path){
```

Comment:
Itt egymás után több függvénynek is az eleje majdnem tökéletesen ugyanaz. Ezeket a redundáns részeket el kellene tüntetni.